include/mbedtls/x509_crt.h

```diff
@@ -97,6 +99,44 @@ typedef struct mbedtls_x509_crt
 }
 mbedtls_x509_crt;
 
+/*
+ * From RFC 5280 section 4.2.1.6:
+ * OtherName ::= SEQUENCE {
+ *      type-id    OBJECT IDENTIFIER,
+ *      value      [0] EXPLICIT ANY DEFINED BY type-id }
+ */
+typedef struct mbedtls_x509_san_other_name
+{
+    mbedtls_x509_buf type_id;                   /**< The type id. */
```

Comment:
Minor: Logically, this indexes the union to come, which should be reflected in the comment.
Usually an indexing field is an enumeration, but here it's a raw OID buffer. Can we add
some additional information for users which describes how they can use `type_id` to index the
`value` union?